packages/dd-trace/test/telemetry/index.spec.js

```diff
@@ -14,6 +14,85 @@ const DEFAULT_HEARTBEAT_INTERVAL = 60000
 
 let traceAgent
 
+describe('telemetry (proxy)', () => {
+  let telemetry
+  let proxy
+
+  beforeEach(() => {
+    telemetry = sinon.spy({
+      start () {},
+      stop () {},
+      updateIntegrations () {},
+      updateConfig () {},
+      appClosing () {}
+    })
+
+    proxy = proxyquire('../../src/telemetry', {
+      './telemetry': telemetry
+    })
+  })
+
+  it('should be noop when disabled', () => {
+    proxy.start()
+    proxy.updateIntegrations()
+    proxy.updateConfig([])
+    proxy.appClosing()
+    proxy.stop()
+
+    expect(telemetry.start).to.not.have.been.called
+    expect(telemetry.updateIntegrations).to.not.have.been.called
+    expect(telemetry.updateConfig).to.not.have.been.called
+    expect(telemetry.appClosing).to.not.have.been.called
+    expect(telemetry.stop).to.not.have.been.called
+  })
+
+  it('should proxy when enabled', () => {
+    const config = { telemetry: { enabled: true } }
+
+    proxy.start(config)
+    proxy.updateIntegrations()
+    proxy.updateConfig([], config)
+    proxy.appClosing()
+    proxy.stop()
+
+    expect(telemetry.start).to.have.been.calledWith(config)
+    expect(telemetry.updateIntegrations).to.have.been.called
+    expect(telemetry.updateConfig).to.have.been.called
+    expect(telemetry.appClosing).to.have.been.called
+    expect(telemetry.stop).to.have.been.called
+  })
+
+  it('should be noop when disabled after being enabled', () => {
+    const config = { telemetry: { enabled: true } }
+
+    proxy.start(config)
+    proxy.stop()
+
+    config.telemetry.enabled = false
+
+    proxy.start(config)
+    proxy.updateIntegrations()
+    proxy.updateConfig([], config)
+    proxy.appClosing()
+    proxy.stop()
+
+    expect(telemetry.start).to.have.been.calledOnce
+    expect(telemetry.updateIntegrations).to.not.have.been.called
+    expect(telemetry.updateConfig).to.not.have.been.called
+    expect(telemetry.appClosing).to.not.have.been.called
+    expect(telemetry.stop).to.have.been.calledOnce
+  })
+
+  it('should proxy when config is updated', () => {
+    const config = { telemetry: { enabled: true } }
+    const changes = []
+
+    proxy.updateConfig(changes, config)
+
+    expect(telemetry.updateConfig).to.have.been.calledWith(changes, config)
```

Comment:
Please make the test more robust by checking for the whole telemetry object being replaced

Reply:
Turned out that test and its underlying functionality were not even needed to begin with.